### PR TITLE
Changed code to improve output for files under test/e2e/framework

### PR DESCRIPTION
**Pull request by NikhilSharmaWe**
#### What type of PR is this?
Enhancement

#### What this PR does / why we need it:
For better (more informative) output for developers when test fails. Changed files are under test/e2e/framework for this PR.
#### Which issue(s) this PR fixes:

Part of #105678 

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -1377,7 +1377,9 @@ retriesLoop:
 		// NOTE the test may need access to the events to see what's going on, such as a change in status
 		actualWatchEvents := scenario(resourceWatch)
 		errs := sets.NewString()
-		ExpectEqual(len(expectedWatchEvents) <= len(actualWatchEvents), true, "Error: actual watch events amount (%d) must be greater than or equal to expected watch events amount (%d)", len(actualWatchEvents), len(expectedWatchEvents))
+		if (!(len(expectedWatchEvents) <= len(actualWatchEvents)) {
```
Comment (oomichi) [inline test/e2e/framework/util.go:1380]: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/105939/pull-kubernetes-verify/1453650290241703936
```
test/e2e/framework/util.go:1380:61: syntax error: unexpected {, expecting )
test/e2e/framework/util.go:1381:4: syntax error: unexpected framework, expecting {
test/e2e/framework/util.go:1382:3: syntax error: unexpected }, expecting expression
```
one more `)` seems necesary after `actualWatchEvents)` for `!(`.
Or just `if (len(expectedWatchEvents) > len(actualWatchEvents)) {`

Comment (pohly) [reply test/e2e/framework/util.go:1380]: ... and drop the unnecessary brackets. This is Go, not C :grin:

**Comment by NikhilSharmaWe** [comment]
I think "k8s.io/kubernetes/test/e2e/framework" should be imported. Am I correct on this?

**Comment by NikhilSharmaWe** [comment]
I am importing "k8s.io/kubernetes/test/e2e/framework" because one of the test showed that framework is undefined. Please inform if I am wrong on this.

**Comment by NikhilSharmaWe** [comment]
This error is showing after I imported "k8s.io/kubernetes/test/e2e/framework".

_package k8s.io/kubernetes/test/e2e
	imports k8s.io/kubernetes/test/e2e/common
	imports k8s.io/kubernetes/test/e2e/common/network
	imports k8s.io/kubernetes/test/e2e/framework
	imports k8s.io/kubernetes/test/e2e/framework: import cycle not allowed_

File: test/e2e/framework/util.go [hunk 2]
```diff
@@ -64,6 +64,7 @@ import (
 	imageutils "k8s.io/kubernetes/test/utils/image"
 	uexec "k8s.io/utils/exec"
 	netutils "k8s.io/utils/net"
+	"k8s.io/kubernetes/test/e2e/framework"
```
Comment (pohly) [inline test/e2e/framework/util.go:67]: This is unnecessary. A package doesn't need to import itself.

File: test/e2e/framework/util.go [hunk 3]
```diff
@@ -1377,7 +1378,9 @@ retriesLoop:
 		// NOTE the test may need access to the events to see what's going on, such as a change in status
 		actualWatchEvents := scenario(resourceWatch)
 		errs := sets.NewString()
-		ExpectEqual(len(expectedWatchEvents) <= len(actualWatchEvents), true, "Error: actual watch events amount (%d) must be greater than or equal to expected watch events amount (%d)", len(actualWatchEvents), len(expectedWatchEvents))
+		if len(expectedWatchEvents) > len(actualWatchEvents) {
+			framework.Failf("Error: actual watch events amount", len(actualWatchEvents), "must be greater than or equal to expected watch events", len(expectedWatchEvents))
```
Comment (pohly) [inline test/e2e/framework/util.go:1382]: Inside the "framework" package itself, `Failf` can be invoked without the `framework.` prefix.

File: test/e2e/framework/util.go [hunk 4]
```diff
@@ -1377,7 +1378,9 @@ retriesLoop:
 		// NOTE the test may need access to the events to see what's going on, such as a change in status
 		actualWatchEvents := scenario(resourceWatch)
 		errs := sets.NewString()
-		ExpectEqual(len(expectedWatchEvents) <= len(actualWatchEvents), true, "Error: actual watch events amount (%d) must be greater than or equal to expected watch events amount (%d)", len(actualWatchEvents), len(expectedWatchEvents))
+		if len(expectedWatchEvents) > len(actualWatchEvents) {
```
Comment (pohly) [inline test/e2e/framework/util.go:1381]: This is a case for `Expect(len(expectedWatchEvents)).To(BeNumerically(">", len(actualWatchEvents), ...)`

Comment (NikhilSharmaWe) [reply test/e2e/framework/util.go:1380]: Removed the import statement but why in the error it was showing that framework is undefined when the import statement was not present.

**Comment by NikhilSharmaWe** [comment]
@pohly One of the test is saying that 
`# k8s.io/kubernetes/test/e2e/framework
test/e2e/framework/util.go:1381:9: Failf call has arguments but no formatting directives
# k8s.io/kubernetes/test/e2e/framework/providers/kubemark
test/e2e/framework/providers/kubemark/kubemark.go:79:19: Failf call has arguments but no formatting directives`
**so, should I add %v format directive in the Failf function.**

File: test/e2e/framework/util.go (hunk 4 above)
Comment (NikhilSharmaWe) [reply test/e2e/framework/util.go:1380]: Removed framework. prefix

**Comment by NikhilSharmaWe** [comment]
@pohly and @oomichi  One of the test is saying that 
` k8s.io/kubernetes/test/e2e/framework
test/e2e/framework/util.go:1381:9: Failf call has arguments but no formatting directives
 k8s.io/kubernetes/test/e2e/framework/providers/kubemark
test/e2e/framework/providers/kubemark/kubemark.go:79:19: Failf call has arguments but no formatting directives`
**so, should I add %v format directive in the Failf function.**

**Comment by NikhilSharmaWe** [comment]
Will this output will be fine 
if !errs.Len() > 0 {
			Failf("Error : %v", strings.Join(errs.List(), "\n - "))
		}
at place of
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L1409

**Comment by NikhilSharmaWe** [comment]
@pohly if you can confirm that the output for all cases are fine, then I can convert the changes to a single commit.

**Comment by NikhilSharmaWe** [comment]
@pohly and @oomichi if you can confirm that the output for all cases are fine, then I can convert the changes to a single commit.

File: test/e2e/framework/util.go (hunk 4 above)
Comment (pohly) [reply test/e2e/framework/util.go:1380]: The entire `if Failf` can be replaced with:
```
Expect(len(expectedWatchEvents)).To(BeNumerically("<=", len(actualWatchEvents), "Did not get enough watch events")
```

File: test/e2e/framework/util.go [hunk 5]
```diff
@@ -1406,7 +1408,9 @@ retriesLoop:
 			fmt.Println("invariants violated:\n", strings.Join(errs.List(), "\n - "))
 			continue retriesLoop
 		}
-		ExpectEqual(errs.Len() > 0, false, strings.Join(errs.List(), "\n - "))
+		if errs.Len() > 0 {
+			Failf("Error: %v", strings.Join(errs.List(), "\n - "))
```
Comment (pohly) [inline test/e2e/framework/util.go:1412]:
```suggestion
			Failf("Unexpected error(s): %v", strings.Join(errs.List(), "\n - "))
```

**Review by pohly — CHANGES_REQUESTED** [review]

**Comment by pohly** [comment]
> then I can convert the changes to a single commit

Feel free to do that right away before pushing an update. At least I prefer it that way (avoids one roundtrip when the result is okay).

**Comment by NikhilSharmaWe** [comment]
@pohly don't you think that "github.com/onsi/gomega" should be imported.

File: test/e2e/framework/util.go (hunk 5 above)
Comment (NikhilSharmaWe) [reply test/e2e/framework/util.go]: gomega.Expect will be used don't you think

**Comment by NikhilSharmaWe** [comment]
gomega.Expect will be used don't you think

**Review by pohly — APPROVED** [review]
/lgtm
/approve

**Comment by NikhilSharmaWe** [comment]
@oomichi could you please sponsor my application for becoming a member of Kubernetes organization.

**Comment by oomichi** [comment]
> 
> 
> @oomichi could you please sponsor my application for becoming a member of Kubernetes organization.

According to https://k8s.devstats.cncf.io/d/13/developer-activity-counts-by-repository-group?orgId=1[…]var-country_name=All&var-repo_name=kubernetes%2Fkubernetes
you already have good contributions on this develop cycle.
I am happy to support you, please let me know if you open an issue for the organization membership.

**Pull request merged**